compiler-core/src/type_/environment.rs

```diff
@@ -497,11 +501,15 @@ impl<'a> Environment<'a> {
             .entity_usages
             .pop()
             .expect("Expected a bottom level of entity usages.");
-        self.handle_unused(unused);
+        self.handle_unused(unused.clone());
 
         for (name, location) in self.unused_modules.clone().into_iter() {
-            self.warnings
-                .emit(Warning::UnusedImportedModule { name, location });
+            let warning = if self.imported_module_names.contains(&name) {
+                Warning::UnusedImportedModuleName { name, location }
```

Comment:
Took me quite a while to understand this warning and the `imported_module_names`. They're not module names, they're module aliases. A module always has a name, but only sometimes has an alias. Let's rename these.

Reply:
Oops, sorry about that, module aliases looks better, thanks!